src/lib.rs

```diff
@@ -1807,7 +1807,22 @@ fn parse_seek_head<R: Read + Seek>(
                 if let ElementData::Location { offset, size } = entry_data {
                     let seek_fields = collect_children(&mut file, *offset, *size)?;
                     if let Ok(seek_entry) = SeekEntry::new(&mut file, &seek_fields) {
-                        seek_head.insert(seek_entry.id, segment_data_offset + seek_entry.offset);
+                        if seek_entry.id == ElementId::SeekHead {
+                            file.seek(SeekFrom::Start(segment_data_offset + seek_entry.offset))?;
+                            let (_element_id, element_data) = next_element(file)?;
+                            assert!(_element_id == ElementId::SeekHead);
+                            //println!("SeekHead : ID : {element_id:?} Data : `{element_data:?}`");
+                            if let ElementData::Location { offset, size } = element_data {
+                                // Push next SeekHead Element
+                                let _old = optional_seek_head.replace((offset, size));
+                                assert!(_old.is_none());
```

Comment:
Again, is this really unrecoverable?

Reply:
It seems to me that shouldn't happen, multiple SeekHead seems to be organized as linked list and result more from an programming issue.
But I supposed an error could be appropriate.

Reply:
Yes, this is a hard error then. But no need to assert as I wrote above.

Reply:
expect is for `Some(_)` result, and there is no `expect_none` unlike `expect_err` for Result.
I have miss something ?

Reply:
You're right. Let's then just check it with "debug_assert" at least, so that release builds don't panic in production because we messed up.